### DISCUSSION TITLE
how to genreate uuid while rollback the data

### DISCUSSION THREAD

**Discussion by chenshi5012** [minio/minio#17303] [Q&A] [answered]
HI，@klauspost @harshavardhana @vadmeste @aead @vadmeste  
Thanks  

I follow your advice： https://github.com/minio/minio/pull/11758  to use [xl-meta-tools ](https://github.com/minio/minio/blob/master/docs/debugging/xl-meta/main.go)  decode "xl.meta",  thanks very much . it works.

![image](https://github.com/minio/minio/assets/10794822/2b5fa4f5-fd2f-422d-b518-4ec6f9f7e86a)

i try to restore "xl.meta" => "xl.meta"+"part.1"

i got the json format xl.meta and the data file 
`
{
  "Versions": [
    {
      "Type": 1,
      "V2Obj": {
        "CSumAlgo": 1,
        "DDir": "n2w74kY/R3GmxxIEuqo/8g==",
        "EcAlgo": 1,
        "EcBSize": 1048576,
        "EcDist": [
          2,
          3,
          4,
          1
        ],
        "EcIndex": 2,
        "EcM": 2,
        "EcN": 2,
        "ID": "AAAAAAAAAAAAAAAAAAAAAA==",
        "MTime": 1652961658145679000,
        "MetaSys": {
          "x-minio-internal-inline-data": "dHJ1ZQ=="
        },
        "MetaUsr": {
          "content-type": "application/octet-stream",
          "etag": "fffb6377948ebea75ad2b8058e849ef5"
        },
        "PartASizes": [
          132096
        ],
        "PartETags": [
          ""
        ],
        "PartNums": [
          1
        ],
        "PartSizes": [
          132096
        ],
        "Size": 132096
      }
    }
  ]
}`

I wonder the document  "9f6c3be2-463f-4771-a6c7-1204baaa3ff2",  how to generate it  ,  is there any  Special  Rules to generate the uuid, or just ordinary UUID.

**Comment by klauspost** [comment, marked as answer]
`uuid_encode(base64_decode(DDIR))`

https://go.dev/play/p/ZGebmQ2dauJ

(DDIR is not actually base64 in metadata, that is the JSON export that does that).

**Reply by chenshi5012** [reply]
thanks， You are the god in my heart
